ws-client/src/transport.rs

```diff
@@ -227,7 +233,24 @@ impl<'a> WsTransportClientBuilder<'a> {
 					match self.mode {
 						Mode::Plain => TlsOrPlain::Plain(socket),
 						Mode::Tls => {
-							let connector = async_tls::TlsConnector::default();
+							let mut client_config = rustls::ClientConfig::default();
+							if let Some(path) = &self.custom_certificate {
+								// TODO: How to give use async for file io and still give add_pem_file() a std::io::BufRead?
+								let f = std::fs::File::open(&path)?;
```

Comment:
Is there any way to solve this async:ier?

[`add_pem_file`](https://docs.rs/rustls/0.19.1/rustls/struct.RootCertStore.html#method.add_pem_file) wants an `std::io::BufRead`.

Reply:
async_std::fs::read(path).await?;